solidity/contracts/LPRewards.sol

```diff
@@ -0,0 +1,211 @@
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+*/
+
+pragma solidity 0.5.17;
+
+import "openzeppelin-solidity/contracts/math/Math.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/SafeERC20.sol";
+
+contract IRewardDistributionRecipient is Ownable {
+    address rewardDistribution;
+
+    function notifyRewardAmount(uint256 reward) external;
+
+    modifier onlyRewardDistribution() {
+        require(
+            msg.sender == rewardDistribution,
+            "Caller is not reward distribution"
+        );
+        _;
+    }
+
+    function setRewardDistribution(address _rewardDistribution)
+        external
+        onlyOwner
+    {
+        rewardDistribution = _rewardDistribution;
+    }
+}
+
+contract LPTokenWrapper {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    uint256 private _totalSupply;
+    mapping(address => uint256) private _balances;
+
+    IERC20 public wrapped; // LPToken pair KEEP/ETH, TBTC/ETH, KEEP/TBTC
+
+    constructor(IERC20 _wrapped) public {
+        wrapped = _wrapped;
+    }
+
+    function totalSupply() public view returns (uint256) {
+        return _totalSupply;
+    }
+
+    function balanceOf(address account) public view returns (uint256) {
+        return _balances[account];
+    }
+
+    function stake(uint256 amount) public {
+        _totalSupply = _totalSupply.add(amount);
+        _balances[msg.sender] = _balances[msg.sender].add(amount);
+        wrapped.safeTransferFrom(msg.sender, address(this), amount);
+    }
+
+    function withdraw(uint256 amount) public {
+        _totalSupply = _totalSupply.sub(amount);
+        _balances[msg.sender] = _balances[msg.sender].sub(amount);
+        wrapped.safeTransfer(msg.sender, amount);
+    }
+}
+
+contract LPRewards is LPTokenWrapper, IRewardDistributionRecipient {
+    IERC20 public rewardToken; // KEEP
```

Comment:
If it's always KEEP - even the comment says that - why don't we hardcode it?

EDIT: OK, I see - tests.

Reply:
So maybe we could name the field `keepToken`?

Reply:
done in 85819dc